plugins/inputs/icinga2/icinga2_test.go

```diff
@@ -1,120 +1,315 @@
-            "type": "Service"
-        }
-    ]
+func TestIcinga2Default(t *testing.T) {
+	// This test should succeed with the default initialization.
+	icinga2 := &Icinga2{
+		Server:          "https://localhost:5665",
+		Objects:         []string{"services"},
+		ResponseTimeout: config.Duration(time.Second * 5),
+	}
+	require.NoError(t, icinga2.Init())
+
+	require.Equal(t, config.Duration(5*time.Second), icinga2.ResponseTimeout)
+	require.Equal(t, "https://localhost:5665", icinga2.Server)
+	require.Equal(t, []string{"services"}, icinga2.Objects)
 }
-`
 
-	checks := Result{}
-	require.NoError(t, json.Unmarshal([]byte(s), &checks))
+func TestIcinga2DeprecatedHostConfig(t *testing.T) {
+	icinga2 := &Icinga2{
+		ObjectType: "hosts", //deprecated
+		Objects:    []string{},
+	}
+	require.NoError(t, icinga2.Init())
 
-	icinga2 := new(Icinga2)
-	icinga2.Log = testutil.Logger{}
-	icinga2.ObjectType = "services"
-	icinga2.Server = "https://localhost:5665"
+	require.Equal(t, []string{"hosts"}, icinga2.Objects)
+}
 
-	var acc testutil.Accumulator
-	icinga2.GatherStatus(&acc, checks.Results)
+func TestIcinga2DeprecatedServicesConfig(t *testing.T) {
+	icinga2 := &Icinga2{
+		ObjectType: "services", //deprecated
+		Objects:    []string{},
+	}
+	require.NoError(t, icinga2.Init())
 
-	expected := []telegraf.Metric{
-		testutil.MustMetric(
-			"icinga2_services",
-			map[string]string{
-				"display_name":  "eq-par.dc2.fr",
+	require.Equal(t, []string{"services"}, icinga2.Objects)
+}
+
+const icinga2ServiceResponse = `{
+	"results": [
+		{
+			"attrs": {
 				"check_command": "check-bgp-juniper-netconf",
-				"state":         "ok",
-				"source":        "someserverfqdn.net",
-				"server":        "localhost",
-				"port":          "5665",
-				"scheme":        "https",
-			},
-			map[string]interface{}{
-				"name":       "ef017af8-c684-4f3f-bb20-0dfe9fcd3dbe",
-				"state_code": 0,
+				"display_name": "eq-par.dc2.fr",
+				"host_name": "someserverfqdn.net",
+				"name": "ef017af8-c684-4f3f-bb20-0dfe9fcd3dbe",
+				"state": 0
 			},
-			time.Unix(0, 0),
-		),
+			"joins": {},
+			"meta": {},
+			"name": "eq-par.dc2.fr!ef017af8-c684-4f3f-bb20-0dfe9fcd3dbe",
+			"type": "Service"
+		}
+	]
+}
+`
+
+func TestGatherServicesStatus(t *testing.T) {
+	var icinga2 = &Icinga2{
+		Server:  "http://localhost:5665",
+		Objects: []string{"services"},
+	}
+	require.NoError(t, icinga2.Init())
+
+	t.Logf("Starting icinga2 HTTP mock (%s)", icinga2.Server)
+
+	ts := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		_, err := fmt.Fprintf(w, "%s", string(icinga2ServiceResponse))
+		require.NoError(t, err)
+	}))
+
+	requestURL, err := url.Parse(icinga2.Server)
+	require.NoError(t, err)
+	require.NotNil(t, requestURL)
+
+	ts.Listener, err = net.Listen("tcp", fmt.Sprintf("%s:%s", requestURL.Hostname(), requestURL.Port()))
```

Comment:
I would not use a fixed port, this might fail on the CI under certain circumstances.  See for example the tests for the http input.

Reply:
Yes, you're right. I've updated test cases to use random ports.